app/src/main/java/com/futurumgame/base/serialization/parsing/ResourceParseRule.java

```diff
@@ -0,0 +1,53 @@
+package com.futurumgame.base.serialization.parsing;
+
+import com.futurumgame.base.additionalDatatypes.Units;
+import com.futurumgame.base.util.CollectionHelper;
+import com.futurumgame.base.enums.Separator;
+import com.futurumgame.base.exceptions.UnitsFormatException;
+import com.futurumgame.base.resources.Resource;
+import com.futurumgame.base.util.Logger;
+import com.futurumgame.base.util.ResourceMapping;
+
+public class ResourceParseRule extends BaseParseRule<Resource> {
+
+    public ResourceParseRule() {
+    }
+
+    @Override
+    public ParseResult<Resource> next(String string) {
+        super.next(string);
+        if (getReadChars().isEmpty()) {
+            return ParseResult.failResult();
+        }
+        String[] data = getReadChars().split(Separator.DataStructureSeparator.getSeparator());
+        ParseResult<Resource> result = ParseResult.failResult();
+        try {
+            if (data.length == 0) {
```

Comment:
String.split does not return an Array with the length of zero... If the line separator is not found in the String then the zeroth element contains the whole string those making it at least a length of 1.... -> Useless if-statement